### PR TITLE
[FLINK-35981][transform] Transform rule support referencing one column more than once

### PR DESCRIPTION
**Pull request by MOBIN-F**
Currently, transform rule (projection / filtering) doesn't support referencing one column more than once, which means if we write such a projection rule:

transform:

projection: *, age * age AS age_square
filter: age < 18 OR age > 60

```
Caused by: org.apache.flink.shaded.guava31.com.google.common.util.concurrent.UncheckedExecutionException: org.apache.flink.api.common.InvalidProgramException: Expression cannot be compiled. This is a bug. Please file an issue.
Expression: import static org.apache.flink.cdc.runtime.functions.SystemFunctionUtils.*;age * age
```

### CODE REVIEW

**Comment by MOBIN-F** [comment]
please help to review this PR, thanks. @yuxiqian

File: flink-cdc-runtime/src/test/java/org/apache/flink/cdc/runtime/operators/transform/TransformDataOperatorTest.java [hunk 1]
```diff
@@ -504,6 +514,58 @@ void testMetadataASTransform() throws Exception {
...
+                        COLUMN_SQUARE_TABLE, recordDataGenerator.generate(new Object[] {1, 1, 1}));
+
+        DataChangeEvent insertEvent2 =
+                DataChangeEvent.insertEvent(
+                        COLUMN_SQUARE_TABLE,
+                        recordDataGenerator.generate(new Object[] {6, 6, null}));
+        DataChangeEvent insertEventExpect2 =
+                DataChangeEvent.insertEvent(
+                        COLUMN_SQUARE_TABLE, recordDataGenerator.generate(new Object[] {6, 6, 36}));
+
```
Comment (yuxiqian) [inline flink-cdc-runtime/src/test/java/org/apache/flink/cdc/runtime/operators/transform/TransformDataOperatorTest.java:552]: Could you please insert one more event `{4, 4, null}` to verify if the filtering block is working as expected?

**Review by yuxiqian — COMMENTED** [review]
Looks good! Just left a minor comment about test case.

File: flink-cdc-runtime/src/test/java/org/apache/flink/cdc/runtime/operators/transform/TransformDataOperatorTest.java (hunk 1 above)
Comment (MOBIN-F) [reply flink-cdc-runtime/src/test/java/org/apache/flink/cdc/runtime/operators/transform/TransformDataOperatorTest.java:552]: done

**Review by yuxiqian — APPROVED** [review]
Thanks for your contribution! To make CI pass, please wait for #3516 to be merged, and rebase this PR to `master` then.

**Comment by yuxiqian** [comment]
Hi @MOBIN-F, sorry but another rebasing might be required since #3465 got merged.

**Comment by MOBIN-F** [comment]
> Hi @MOBIN-F, sorry but another rebasing might be required since #3465 got merged. :(

done

**Review by leonardBang — APPROVED** [review]
Thanks @MOBIN-F for the contribution, LGTM

**Pull request merged**